API/SelectU.Core/Services/ReviewService.cs

```diff
@@ -0,0 +1,94 @@
+﻿using SelectU.Contracts.DTO;
+using SelectU.Contracts;
+using SelectU.Contracts.Services;
+using SelectU.Core.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using SelectU.Contracts.Entities;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Http.HttpResults;
+
+namespace SelectU.Core.Services
+{
+    public class ReviewService: IReviewService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+
+        public ReviewService(IUnitOfWork context)
+        {
+            _unitOfWork = context;
+        }
+
+
+        public async Task CreateScholarshipApplicationRatingAsync(UserRatingDTO userRatingDTO)
+        {
+            var scholarshipApplication = await _unitOfWork.ScholarshipApplications.GetAsync(userRatingDTO.ScholarshipApplicationId);
+
+            if (scholarshipApplication == null) throw new ReviewException($"Unable able to add rating as the application does not exist");
+
+            if (scholarshipApplication.Ratings.FirstOrDefault(x => x.ReviewerId == userRatingDTO.ReviewerId) == null) throw new ReviewException($"Unable able to create rating as the reviewer has an existing review");
+
+            UserRating userRating = new UserRating()
+            {
+                ScholarshipApplicationId = userRatingDTO.ScholarshipApplicationId,
+                Rating = userRatingDTO.Rating,
+                Id = userRatingDTO.Id,
+                ReviewerId = userRatingDTO.ReviewerId,
+                ApplicantId = userRatingDTO.ApplicantId,
+                Comment = userRatingDTO.Comment
+            };
+
+            scholarshipApplication.Ratings.Add(userRating);
+            _unitOfWork.ScholarshipApplications.Update(scholarshipApplication);
+
+            await _unitOfWork.CommitAsync();
+
+        }
+
+        public async Task UpdateScholarshipApplicationRatingAsync(UserRatingDTO userRatingDTO, bool isAdmin)
+        {
+            var scholarshipApplication = await _unitOfWork.ScholarshipApplications.GetAsync(userRatingDTO.ScholarshipApplicationId);
+
+            if (scholarshipApplication == null)
+            {
+                throw new ReviewException($"Unable able to add rating as the application does not exist");
+            }
+
+            UserRating userRating = new UserRating()
+            {
+                ScholarshipApplicationId = userRatingDTO.ScholarshipApplicationId,
+                Rating = userRatingDTO.Rating,
+                Id = userRatingDTO.Id,
+                ReviewerId = userRatingDTO.ReviewerId,
+                ApplicantId = userRatingDTO.ApplicantId,
+                Comment = userRatingDTO.Comment
+            };
+            var originalRating = scholarshipApplication.Ratings.FirstOrDefault(x => x.Id == userRating.Id && (x.ReviewerId == userRatingDTO.ReviewerId || isAdmin));
```

Comment:
var review= await _unitOfWork.Reviews.GetAsync(reviewDTO.Id));
update the returned entity based on the update dto
_unitOfWork.Reviews.Update(review);